src/SprykerSdk/Zed/ApplicationRoute/Business/Model/Parser/YvesResourceCreatorParser.php

```
@@ -0,0 +1,120 @@
+ */
+
+namespace SprykerSdk\Zed\ApplicationRoute\Business\Model\Parser;
+
+use Exception;
+use Generated\Shared\Transfer\ResourceCreatorListTransfer;
+use Generated\Shared\Transfer\ResourceCreatorTransfer;
+use Symfony\Component\Finder\Finder;
+
+class YvesResourceCreatorParser implements YvesResourceCreatorParserInterface
+{
+    const SEARCH_MODULE_NAME = '~[^\\\]+(?=\\\ResourceCreator)~';
+    const GET_NAMESPACE = '~(?:namespace\s+([^;]++))+~si';
+    const GET_CLASS_NAME = '~(?:class\s+([^\s\n\r]++))+~si';
+    const FK_RESOURCE_ = 'fk_resource_';
+    const GET_TYPE_METHOD = 'getType';
+    const GET_FUNCTION = '~(?:public)?\s+function\s+(((?!Resource).)*Resource)~xm';
+
+    /**
+     * @var \Symfony\Component\Finder\Finder
+     */
+    protected $finder;
+
+    /**
+     * @param \Symfony\Component\Finder\Finder $finder
+     */
+    public function __construct(Finder $finder)
+    {
+        $this->finder = $finder;
+    }
+
+    /**
+     * @param string $resourceCreatorSearchPattern
+     *
+     * @return \Generated\Shared\Transfer\ResourceCreatorListTransfer $resourceCreatorList
+     */
+    public function getResourceCreatorList(string $resourceCreatorSearchPattern)
+    {
+        $this->finder->in($resourceCreatorSearchPattern);
+
+        $resourceCreatorList = new ResourceCreatorListTransfer();
+        foreach ($this->finder as $item) {
+            $resourceCreator = $this->getResourceCreatorInfo($item->getPathname());
+
+            try {
+                include_once $item->getPathname();
+                $fullClassName = $resourceCreator->getNamespace() . '\\' . $resourceCreator->getClassName();
+                $type = call_user_func([$fullClassName, self::GET_TYPE_METHOD]);
+                $resourceCreator->setType($type);
+                $resourceCreator->setColumnName(self::FK_RESOURCE_ . $type);
+                $resourceCreatorList->addResourceCreators($resourceCreator);
+            } catch (Exception $e) {
```

Empty catch with generic exception class is not allowed.